crates/sui-types/src/effects/effects_v1.rs

```diff
@@ -107,6 +107,11 @@ impl TransactionEffectsAPI for TransactionEffectsV1 {
     fn modified_at_versions(&self) -> Vec<(ObjectID, SequenceNumber)> {
         self.modified_at_versions.clone()
     }
+
+    fn lamport_version(&self) -> SequenceNumber {
+        SequenceNumber::lamport_increment(self.modified_at_versions.iter().map(|(_, v)| *v))
```

Comment:
What about the version of any mutated object that's not Immutable? (since we record owner in the output)

Reply:
We can do that, but IIUC a hypothetical v1 (system) transaction that only creates immutable objects would report a zero lamport version, whereas its v2 equivalent would not (I believe it would report 1).  I think that's fine, because I don't think such a transaction exists, and with the transition to V2 coming up, the window of opportunity for one to show up is closing, but I'm also not sure it necessarily buys us much, to change the implementation like that, right?